scipy/special/tests/test_ndtri_exp.py

```diff
@@ -60,13 +60,32 @@ def test_in_interval(self, interval, expected_rtol, uniform_random_points):
         )
 
     def test_extreme(self):
-        assert_func_equal(
-            log_ndtr_ndtri_exp,
-            lambda y: y,
-            [-np.finfo(float).max, -np.finfo(float).min],
-            rtol=1e-12,
-            nan_ok=True
-        )
+        # bigneg is not quite the largest negative double precision value.
+        # Here's why:
+        # The round-trip calculation
+        #    y = ndtri_exp(bigneg)
+        #    bigneg2 = log_ndtr(y)
+        # where bigneg is a very large negative value, would--with infinite
+        # precision--result in bigneg2 == bigneg.  When bigneg is large enough,
+        # y is effectively equal to -sqrt(2)*sqrt(-bigneg), and log_ndtr(y) is
+        # effectively -(y/sqrt(2))**2.  If we use bigneg = np.finfo(float).min,
+        # then by construction, the theoretical value is the most negative
+        # finite value that can be represented with 64 bit float point.  This
+        # means tiny changes in how the computation proceeds can result in the
+        # return value being -inf.  (E.g. changing the constant representation
+        # of 1/sqrt(2) from 0.7071067811865475--which is the value returned by
+        # 1/np.sqrt(2)--to 0.7071067811865476--which is the most accurate 64
+        # bit floating point representation of 1/sqrt(2)--results in the
+        # round-trip that starts with np.finfo(float).min returning -inf.  So
+        # we'll move the bigneg value a few ULPs towards 0 to avoid this
+        # sensitivity.
+        # Use the reduce method to apply nextafter four times.
```

Comment:
This is a really good consideration. I wrote this test kind of as an after-thought and didn't pay sufficient attention that everything actually made sense. Sigh, that `nan_ok` is practically unforgivable. Thanks for the in-depth explanation.

Reply:
I wouldn't worry to much about it - that `nan_ok` has bitten us many a time. I once had ambitions to turn it off everywhere and then remove that flag, but that turned out to be more complex than hoped. A change that flips it to being default off would be super smooth though!